### PR TITLE
nspr: bump + modernize + no tools.os_info in `build_requirements`

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **nspr/1.0**

- bump
- for https://github.com/conan-io/hooks/pull/320

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@timblechmann 
When building nspr for Apple M1, the build script detects `CPU_ARCH=ppc`.
Since you committed https://github.com/conan-io/conan-center-index/pull/4770 (and own a Mac), can you take a look at the apple failure?

`config.log` is printed out in the build logs of https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/7212/2-configs/macos-m1-clang/nspr/4.27//summary.json
It shows `CPU_ARCH='ppc'`

**Comment by timblechmann** [comment]
hi @madebr: unfortunately i don't have access to an m1 (i had only ssh-ed into one for a project last year.

that said: older nspr versions don't build on apple silicon: it is only supported since 4.29. but i guess you could try to backport the patches listed in https://bugzilla.mozilla.org/show_bug.cgi?id=1658671

File: recipes/nspr/all/conanfile.py [hunk 1]
```diff
@@ -31,35 +34,38 @@ class NsprConan(ConanFile):
...
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = os.path.join("nspr-" + self.version, "nspr")
-        os.rename(extracted_dir, self._source_subfolder)
-        tools.rmdir(os.path.join("nspr-" + self.version))
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination="tmp", strip_root=True)
```
Comment (prince-chrismc) [inline recipes/nspr/all/conanfile.py:55]: Feature request? `strip_root=2`

**Review by prince-chrismc — APPROVED** [review]

File: recipes/nspr/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/nspr/all/conanfile.py:55]: It worths an issue

**Comment by jgsogo** [comment]
Something is wrong with this PR. The CI has been running for 4 days. No output so far. We need to check it carefully.

File: recipes/nspr/all/conanfile.py [hunk 2]
```diff
@@ -89,20 +99,18 @@ def _configure_autotools(self):
         if self.settings.compiler == "Visual Studio":
             conf_args.extend([
                 "{}-pc-mingw32".format("x86_64" if self.settings.arch == "x86_64" else "x86"),
-                "--enable-static-rtl" if "MT" in str(self.settings.compiler.runtime) else "--disable-static-rtl",
-                "--enable-debug-rtl" if "d" in str(self.settings.compiler.runtime) else "--disable-debug-rtl",
+                "--enable-static-rtl={}".format(yes_no("MT" in self.settings.compiler.runtime)),
+                "--enable-debug-rtl={}".format(yes_no("d" in self.settings.compiler.runtime)),
```
Comment (SSE4) [inline recipes/nspr/all/conanfile.py:103]:
```suggestion
                "--enable-static-rtl" if "MT" in str(self.settings.compiler.runtime) else "--disable-static-rtl",
                "--enable-debug-rtl" if "d" in str(self.settings.compiler.runtime) else "--disable-debug-rtl",
```

**Review by uilianries — APPROVED** [review]

File: recipes/nspr/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/nspr/all/conanfile.py:65]: please fill a feature request to conan client

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

File: recipes/nspr/all/conanfile.py [hunk 3]
```diff
@@ -79,30 +84,34 @@ def _build_context(self):
...
+            "--with-mozilla={}".format(yes_no(self.options.with_mozilla)),
+            "--enable-64bit={}".format(yes_no(self.settings.arch in ("armv8", "x86_64", "mips64", "ppc64", "ppc64le"))),
+            "--enable-strip={}".format(yes_no(self.settings.build_type not in ("Debug", "RelWithDebInfo"))),
+            "--enable-debug={}".format(yes_no(self.settings.build_type == "Debug")),
+            "--datarootdir={}".format(tools.unix_path(os.path.join(self.package_folder, "res"))),
             "--disable-cplus",
             "--enable-64bit" if self.settings.arch in ("armv8", "x86_64") else "--disable-64bit",
             "--disable-strip" if self.settings.build_type == "RelWithDebInfo" else "--enable-strip",
             "--enable-debug" if self.settings.build_type == "Debug" else "--disable-debug",
         ]
```
Comment (SpaceIm) [inline recipes/nspr/all/conanfile.py:99]: there are dupicated opions here.

File: recipes/nspr/all/conanfile.py [hunk 4]
```diff
@@ -79,30 +84,34 @@ def _build_context(self):
...
-        if self.settings.compiler == "Visual Studio":
+        if self._is_msvc:
             conf_args.extend([
                 "{}-pc-mingw32".format("x86_64" if self.settings.arch == "x86_64" else "x86"),
-                "--enable-static-rtl" if "MT" in str(self.settings.compiler.runtime) else "--disable-static-rtl",
-                "--enable-debug-rtl" if "d" in str(self.settings.compiler.runtime) else "--disable-debug-rtl",
+                "--enable-static-rtl={}".format(yes_no("MT" in self.settings.compiler.runtime)),
+                "--enable-debug-rtl={}".format(yes_no("d" in self.settings.compiler.runtime)),
```
Comment (SpaceIm) [inline recipes/nspr/all/conanfile.py:104]:
```suggestion
                "--enable-static-rtl={}".format(yes_no("MT" in msvc_runtime_flag(self))),
                "--enable-debug-rtl={}".format(yes_no("d" in msvc_runtime_flag(self))),
```

File: recipes/nspr/all/conanfile.py [hunk 5]
```diff
@@ -1,35 +1,39 @@
 from conans import ConanFile, tools, AutoToolsBuildEnvironment
```
Comment (SpaceIm) [inline recipes/nspr/all/conanfile.py:1]:
```suggestion
from conan.tools.microsoft import msvc_runtime_flag
from conans import ConanFile, tools, AutoToolsBuildEnvironment
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/nspr/all/conanfile.py (hunk 3 above)
Comment (SSE4) [reply recipes/nspr/all/conanfile.py:99]: might be a merge artifact

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**